Identity baseUrl not set or only set on multi-tenancy enabled

**Describe the issue:**

<!-- A clear and concise description of what the issue is. -->

The identity base url is not set correctly for some components.

This leads to problems when using GENERIC oidc mode

**Actual behavior:**

<!-- A clear and concise description of what actually happens. -->

When enabling GENERIC oidc mode, Auth to the zeebeGateway fails.

**Expected behavior:**

<!-- A clear and concise description of what you expected to happen. -->

Auth works

**How to reproduce:**

<!--
Steps to reproduce the issue.

If possible add a minimal reproducer code sample in a new repo/branch.
-->

1. Enable GENERIC oidc
2. configure a client for the zeebe gateway
3. install platform
4. Other applications trying to access the zeebe gateway will be rejected

**Logs:**

<!-- If possible add the full logs related to the issue. -->

Zeebe Gateway:
```

SEVERE: Exception while executing runnable io.grpc.internal.ServerImpl$ServerTransportListenerImpl$1HandleServerCall@7082a708
java.lang.IllegalStateException: java.lang.IllegalArgumentException: URI with undefined scheme
        at io.grpc.internal.ServerImpl$ServerTransportListenerImpl$1HandleServerCall.runInternal(ServerImpl.java:617)
        at io.grpc.internal.ServerImpl$ServerTransportListenerImpl$1HandleServerCall.runInContext(ServerImpl.java:603)
        at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
        at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133)
        at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(Unknown Source)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source)
        at java.base/java.util.concurrent.ForkJoinPool.scan(Unknown Source)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)
Caused by: java.lang.IllegalArgumentException: URI with undefined scheme
        at java.net.http/jdk.internal.net.http.common.Utils.newIAE(Unknown Source)
        at java.net.http/jdk.internal.net.http.HttpRequestBuilderImpl.checkURI(Unknown Source)
        at java.net.http/jdk.internal.net.http.HttpRequestBuilderImpl.uri(Unknown Source)
        at java.net.http/jdk.internal.net.http.HttpRequestBuilderImpl.uri(Unknown Source)
        at io.camunda.identity.sdk.impl.rest.RestClient.request(RestClient.java:60)
        at io.camunda.identity.sdk.impl.generic.GenericAuthentication.getPermissions(GenericAuthentication.java:139)
        at io.camunda.identity.sdk.authentication.AbstractAuthentication.verifyToken(AbstractAuthentication.java:215)
        at io.camunda.identity.sdk.authentication.AbstractAuthentication.verifyToken(AbstractAuthentication.java:164)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source)
        at java.base/java.lang.reflect.Method.invoke(Unknown Source)
        at io.camunda.identity.sdk.annotation.AnnotationProcessor.lambda$apply$0(AnnotationProcessor.java:33)
        at jdk.proxy2/jdk.proxy2.$Proxy108.verifyToken(Unknown Source)
        at io.camunda.zeebe.gateway.interceptors.impl.IdentityInterceptor.interceptCall(IdentityInterceptor.java:79)
        at io.grpc.ServerInterceptors$InterceptCallHandler.startCall(ServerInterceptors.java:269)
        at io.grpc.internal.ServerImpl$ServerTransportListenerImpl.startWrappedCall(ServerImpl.java:701)
        at io.grpc.internal.ServerImpl$ServerTransportListenerImpl.access$2200(ServerImpl.java:408)
        at io.grpc.internal.ServerImpl$ServerTransportListenerImpl$1HandleServerCall.runInternal(ServerImpl.java:613)
        ... 9 more
```
**Environment:**

**Please note: Without the following info, it's hard to resolve the issue and probably it will be closed.**

- Platform: not relevant
- Helm CLI version: not relevant
- Chart version: latest
- Values file: 

```yaml
global:
  identity:
    auth:
      issuer: https://<keycloak base url>/realms/camunda-platform
      issuerBackendUrl: https://<keycloak base url>/realms/camunda-platform
      tokenUrl: https://<keycloak base url>/realms/camunda-platform/protocol/openid-connect/token
      jwksUrl: https://<keycloak base url>/realms/camunda-platform/protocol/openid-connect/certs
      type: "GENERIC"
      identity:
        clientId: identity
        audience: identity
        existingSecret: <identity client secret>
        initialClaimName: sub
        initialClaimValue: <uuid of the first user to access identity and leave all others in>
        redirectUrl: <identity base url>
      operate:
        clientId: operate
        audience: operate
        existingSecret: <operate client secret>
        redirectUrl: <operate base url>
      tasklist:
        clientId: tasklist
        audience: tasklist
        existingSecret: <tasklist client secret>
        redirectUrl: <tasklist base url>
      optimize:
        clientId: optimize
        audience: optimize
        existingSecret: <optimize client secret>
        redirectUrl: <optimize base url>
      zeebe:
        clientId: zeebe
        audience: zeebe
        existingSecret: <zeebe client secret>
      connectors:
        clientId: connectors
        audience: connectors
        existingSecret: <connectors client secret>

# disable the dependency keycloak

identityKeycloak:
  enabled: false

# enable the identity database (unless you have a dedicated database for identity to work with)

identityPostgresql:
  enabled: true
```

--- a/charts/camunda-platform-latest/templates/operate/configmap.yaml
+++ b/charts/camunda-platform-latest/templates/operate/configmap.yaml
@@ -31,6 +31,7 @@ data:
       identity:
         clientId: {{ include "operate.authClientId" . | quote }}
         audience: {{ include "operate.authAudience" . | quote }}
+        baseUrl: {{ include "camundaPlatform.identityURL" . | quote }}
     {{- else }}
     spring:
       profiles:

--- a/charts/camunda-platform-latest/templates/optimize/configmap.yaml
+++ b/charts/camunda-platform-latest/templates/optimize/configmap.yaml
@@ -63,6 +63,8 @@ data:
         audience: {{ include "optimize.authAudience" . | quote }}
         issuer: {{ include "camundaPlatform.authIssuerUrl" . | quote }}
         issuerBackendUrl: {{ include "camundaPlatform.authIssuerBackendUrl" . | quote }}
+        baseUrl: {{ include "camundaPlatform.identityURL" . | quote }}
+
   {{- end }}
 
   {{- range $key, $val := .Values.optimize.extraConfiguration }}

--- a/charts/camunda-platform-latest/templates/tasklist/configmap.yaml
+++ b/charts/camunda-platform-latest/templates/tasklist/configmap.yaml
@@ -32,6 +32,8 @@ data:
       identity:
         clientId: {{ include "tasklist.authClientId" . | quote }}
         audience: {{ include "tasklist.authAudience" . | quote }}
+        baseUrl: {{ include "camundaPlatform.identityURL" . | quote }}
+
     {{- else }}
     spring:
       profiles:

--- a/charts/camunda-platform-latest/templates/zeebe-gateway/configmap.yaml
+++ b/charts/camunda-platform-latest/templates/zeebe-gateway/configmap.yaml
@@ -43,8 +43,6 @@ data:
         type: {{ include "camundaPlatform.authType" . | quote }}
         issuerBackendUrl: {{ include "camundaPlatform.authIssuerBackendUrl" . | quote }}
         audience: {{ include "zeebe.audience" . | quote }}
-        {{- end }}
-        {{- if .Values.global.multitenancy.enabled }}
         baseUrl: {{ include "camundaPlatform.identityURL" . | quote }}
         {{- end }}
     {{- end }}

--- a/charts/camunda-platform-latest/test/unit/operate/golden/configmap.golden.yaml
+++ b/charts/camunda-platform-latest/test/unit/operate/golden/configmap.golden.yaml
@@ -28,6 +28,7 @@ data:
       identity:
         clientId: "operate"
         audience: "operate-api"
+        baseUrl: "http://camunda-platform-test-identity:80"
     
     # Operate configuration file
     camunda.operate:

--- a/charts/camunda-platform-latest/test/unit/tasklist/golden/configmap.golden.yaml
+++ b/charts/camunda-platform-latest/test/unit/tasklist/golden/configmap.golden.yaml
@@ -28,6 +28,7 @@ data:
       identity:
         clientId: "tasklist"
         audience: "tasklist-api"
+        baseUrl: "http://camunda-platform-test-identity:80"
 
     # Tasklist configuration file
     camunda.tasklist:

--- a/charts/camunda-platform-latest/test/unit/zeebe-gateway/golden/configmap-log4j2.golden.yaml
+++ b/charts/camunda-platform-latest/test/unit/zeebe-gateway/golden/configmap-log4j2.golden.yaml
@@ -31,6 +31,7 @@ data:
         type: "KEYCLOAK"
         issuerBackendUrl: "http://camunda-platform-test-keycloak:80/auth/realms/camunda-platform"
         audience: "zeebe-api"
+        baseUrl: "http://camunda-platform-test-identity:80"
     zeebe:
       gateway:
         security:

--- a/charts/camunda-platform-latest/test/unit/zeebe-gateway/golden/configmap.golden.yaml
+++ b/charts/camunda-platform-latest/test/unit/zeebe-gateway/golden/configmap.golden.yaml
@@ -29,6 +29,7 @@ data:
         type: "KEYCLOAK"
         issuerBackendUrl: "http://camunda-platform-test-keycloak:80/auth/realms/camunda-platform"
         audience: "zeebe-api"
+        baseUrl: "http://camunda-platform-test-identity:80"
     zeebe:
       gateway:
         security:

